include/ginkgo/core/base/exception_helpers.hpp

```diff
@@ -757,16 +757,16 @@ inline T ensure_allocated_impl(T ptr, const std::string& file, int line,
 
 /**
  * Ensures that two dimensions have compatible bounds, in particular before a
- * copy operation. This means the target should have at least as much elements
+ * copy operation. This means the target should have the same number of elements
  * as the source.
  *
  * @param _source  the source of the expected copy operation
  * @param _target  the destination of the expected copy operation
  *
- * @throw OutOfBoundsError  if `_source > _target`
+ * @throw OutOfBoundsError  if `_source != _target`
  */
 #define GKO_ENSURE_COMPATIBLE_BOUNDS(_source, _target)                       \
```

Comment:
I dont think the macro should change, but the place where you were using it should use `GKO_ASSERT_EQ` instead. I think this macro is doing the correct thing. 

Actually, looks like this macro is not needed at all. We can consider removing it.